crates/noirc_evaluator/src/ssa/node.rs

```diff
@@ -0,0 +1,948 @@
+        Operation::jne | Operation::jeq | Operation::jmp => false,
+        Operation::phi => false,
+        Operation::nop => false,
+        Operation::eq_gate => true,
+    }
+}
+
+pub fn to_operation(op_kind: HirBinaryOpKind, op_type: ObjectType) -> Operation {
+    match op_kind {
+        HirBinaryOpKind::Add => Operation::add,
+        HirBinaryOpKind::Subtract => Operation::sub,
+        HirBinaryOpKind::Multiply => Operation::mul,
+        HirBinaryOpKind::Equal => Operation::eq,
+        HirBinaryOpKind::NotEqual => Operation::ne,
+        HirBinaryOpKind::And => Operation::and,
+        HirBinaryOpKind::Or => Operation::or,
+        HirBinaryOpKind::Xor => Operation::xor,
+        HirBinaryOpKind::Divide => {
+            if op_type.is_signed() {
+                return Operation::sdiv;
+            }
+            if op_type.is_unsigned() {
+                return Operation::udiv;
+            }
+            if op_type.is_field() {
+                return Operation::div;
+            }
+            unreachable!("invalid type"); //TODO error
+        }
+        HirBinaryOpKind::Less => {
+            if op_type.is_signed() {
+                return Operation::slt;
+            }
+            if op_type.is_unsigned() {
+                return Operation::ult;
+            }
+            if op_type.is_field() {
+                return Operation::lt;
+            }
+            unreachable!("invalid type"); //TODO error
+        }
+        HirBinaryOpKind::Greater => {
+            if op_type.is_signed() {
+                return Operation::sgt;
+            }
+            if op_type.is_unsigned() {
+                return Operation::ugt;
+            }
+            if op_type.is_field() {
+                return Operation::gt;
+            }
+            unreachable!("invalid type"); //TODO error
+        }
+        HirBinaryOpKind::LessEqual => {
+            if op_type.is_signed() {
+                return Operation::sle;
+            }
+            if op_type.is_unsigned() {
+                return Operation::ule;
+            }
+            if op_type.is_field() {
+                return Operation::lte;
+            }
+            unreachable!("invalid type"); //TODO error
+        }
+        HirBinaryOpKind::GreaterEqual => {
+            if op_type.is_signed() {
+                return Operation::sge;
+            }
+            if op_type.is_unsigned() {
+                return Operation::uge;
+            }
+            if op_type.is_field() {
+                return Operation::gte;
+            }
+            unreachable!("invalid type"); //TODO error
+        }
+        HirBinaryOpKind::Assign => Operation::ass,
+        HirBinaryOpKind::MemberAccess => todo!(),
+    }
+}
+
+pub fn get_witness_from_object(obj: &Object) -> Option<Witness> {
+    match obj {
+        Object::Integer(i) => Some(i.witness),
+        Object::Array(_) => todo!(),
+        Object::Constants(_) => None,
+        _ => obj.witness(), // These will
```

Comment:
This comment seems to be unfinished?

Reply:
I think it would be `unreachable!` . This would be the safest option if unsure because it would also cause the compiler to panic leading to further investigation

Reply:
Done

Reply:
I had to add it back because the function was getting called with a (unit) Linear in some case. I am not sure who created it (and why) because the previous evaluator is not called anymore but at least it is working fine with the obj.witness().